tests/Unit/Models/CommentTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Tests\Unit\Models;
+
+use Tests\TestCase;
+use App\Models\Comment;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\MorphMany;
+
+class CommentTest extends TestCase
+{
+    /**
+     * A basic unit test example.
+     *
+     * @return void
+     */
+    protected $comment;
+
+    public function setUp() : void 
+    {
+        $this->comment = new Comment();
+        parent::setUp();
+    }
+
+    public function tearDown() : void 
+    {
+        $this->comment = null;
+        parent::tearDown();
+    }
+
+    public function test_comment_contains_valid_fillable_properties()
+    {
+        $this->assertEquals([
+            'user_id',
+            'post_id',
+            'content',
+        ], $this->comment->getFillable());
+    }
+
+    public function test_user_relationship()
+    {
+        $user = $this->comment->user();
+        $this->assertInstanceOf(BelongsTo::class, $user);
+        $this->assertEquals('user_id', $user->getForeignKeyName());
+        $this->assertEquals('id', $user->getOwnerKeyName());
+    }
+
+    public function test_posts_relationship()
+    {
+        $post = $this->comment->post();
+        $this->assertInstanceOf(BelongsTo::class, $post);
+        $this->assertEquals('post_id', $post->getForeignKeyName());
+        $this->assertEquals('id', $post->getOwnerKeyName());
+    }
+
+    public function test_activities_relationship()
+    {
+        $actable = $this->comment->activities();
+        $this->assertInstanceOf(MorphMany::class, $actable);
+        $this->assertEquals('actable_id', $actable->getForeignKeyName());
+        $this->assertEquals('actable_type', $actable->getMorphType());
+    }
```

Comment:
giải thích các function này

Reply:
- test_user_relationship():
        // lấy 1 instance vừa khởi tạo từ setup gọi đến quan hệ với user
        $user = $this->comment->user();
        // cho cái $user này thuộc về quan hệ BeLongsTo
        $this->assertInstanceOf(BelongsTo::class, $user);
        // so sánh khóa phụ của comment có phải id của user không.
        $this->assertEquals('user_id', $user->getForeignKeyName());
        // so sánh id có bằng khóa chính của user không.
        $this->assertEquals('id', $user->getOwnerKeyName());

- test_posts_relationship():
         // lấy 1 instance vừa khởi tạo từ setup gọi đến quan hệ với post
        $post = $this->comment->post();
        // cho nó thuộc về quan hệ BeLongsTo
        $this->assertInstanceOf(BelongsTo::class, $post);
        // so sánh khóa phụ của post_id của comments có bằng id của post không
        $this->assertEquals('post_id', $post->getForeignKeyName());
         // so sánh id có bằng khóa chính của post không.
        $this->assertEquals('id', $post->getOwnerKeyName());
-  test_activities_relationship();
        // lấy 1 instance vừa khởi tạo từ setup gọi đến quan hệ với activities.
        $actable = $this->comment->activities();
        // cho nó thuộc về quan hệ MorphMany
        $this->assertInstanceOf(MorphMany::class, $actable);
        // so sánh actable_id có bằng id của model
        $this->assertEquals('actable_id', $actable->getForeignKeyName());
        // so sánh actable_type có bằng tên model
        $this->assertEquals('actable_type', $actable->getMorphType());